sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -210,16 +213,30 @@ public <T> T insert(String containerName, T objectToSave) {
      * @return the inserted item
      */
     public <T> T insert(String containerName, T objectToSave, PartitionKey partitionKey) {
+
         Assert.hasText(containerName, "containerName should not be null, empty or only whitespaces");
         Assert.notNull(objectToSave, "objectToSave should not be null");
 
         @SuppressWarnings("unchecked") final Class<T> domainType = (Class<T>) objectToSave.getClass();
-        containerName = getContainerName(domainType);
 
         markAuditedIfConfigured(objectToSave);
         generateIdIfNullAndAutoGenerationEnabled(objectToSave, domainType);
 
-        final JsonNode originalItem = mappingCosmosConverter.writeJsonNode(objectToSave);
+        JsonNode originalItem = mappingCosmosConverter.writeJsonNode(objectToSave);
+        JsonNode preStrippedItem = originalItem.deepCopy();
```

Comment:
We can also skip the deep copy operation in that case and just remove the fields marked transient when mapping them to json node.

Reply:
I did consider using MappingCosmosConverter, the preStrippedItem is created so that the transient fields and their values can later be added back to the response that is returned from each API call (which will have null for each field). If the logic is moved to writeJsonNode, could not see an easy way to add back the values.

Reply:
@TheovanKraay @kushagraThapar still not my favorite but what do you think of this? I ran the test testInsertDocShouldNotPersistTransientFields() you added in CosmosTemplateIT and it was still passing. This still might need some cleanup and needs to be done in all the other places too but I think it might be a pretty good solution.

https://github.com/Azure/azure-sdk-for-java/pull/39810/commits/df00c3df18b800c37b69ae38ed6b3dafee9d22f2

Reply:
I am re-factoring, will update soon....

Reply:
I've refactored based on original code. I preferred to add an overload to writeJsonNode() to switch between stripping transient fields, so that repopulate can be done using JsonNode instead of ObjectNode. But otherwise it is mostly the same approach as @trande4884, all of the logic is now in MappingCosmosConverter. @kushagraThapar what do you think?